jupyterhub/handlers/base.py

```diff
@@ -1107,11 +1107,22 @@ def template_namespace(self):
             logout_url=self.settings['logout_url'],
             static_url=self.static_url,
             version_hash=self.version_hash,
+            services=self.get_accessible_services(user),
         )
         if self.settings['template_vars']:
             ns.update(self.settings['template_vars'])
         return ns
 
+    def get_accessible_services(self, user):
+        accessible_services = list()
+        for service in self.services.values():
+            if not service.url:
+                continue
+            if service.admin and not user.admin:
```

Comment:
This isn't necessarily a correct condition. service.admin means that the service itself has admin permissions (e.g. launching servers for users), not that only admin users can use it. I'd just remove it for now.